src/Microsoft.SqlTools.ServiceLayer/ExecutionPlan/ExecutionPlanService.cs

```diff
@@ -0,0 +1,219 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.ServiceLayer.ExecutionPlan.Contracts;
+using Microsoft.SqlTools.ServiceLayer.ExecutionPlan.ExecPlanGraph;
+using Microsoft.SqlTools.ServiceLayer.ExecutionPlan.ExecPlanGraph.Comparison;
+
+namespace Microsoft.SqlTools.ServiceLayer.ExecutionPlan
+{
+    /// <summary>
+    /// Main class for Migration Service functionality
+    /// </summary>
+    public sealed class ExecutionPlanService : IDisposable
+    {
+        private static readonly Lazy<ExecutionPlanService> instance = new Lazy<ExecutionPlanService>(() => new ExecutionPlanService());
+
+        private bool disposed;
+
+        /// <summary>
+        /// Construct a new MigrationService instance with default parameters
```

Comment:
This should be private if you're using the singleton pattern (and fix the doc comment)